syntax = "proto3";

package multimodal;

service MultimodalService {
  rpc Analyze (MultimodalRequest) returns (MultimodalResponse);
}

message MultimodalRequest {
  string session_id = 1;
  bytes image = 2;
  string audio_path = 3;
  string text = 4;
}

message MultimodalResponse {
  string text_output = 1;
  string detected_language_code = 2;  // ISO language code (e.g., 'en', 'es', 'fr')
  string detected_language_name = 3;  // Language name (e.g., 'English', 'Spanish')
}


